src/compiler/checker.ts

```diff
@@ -6034,11 +6034,14 @@ namespace ts {
             function isKnownProperty(type: Type, name: string): boolean {
                 if (type.flags & TypeFlags.ObjectType) {
                     const resolved = resolveStructuredTypeMembers(type);
-                    if ((relation === assignableRelation || relation === comparableRelation) &&
-                        (type === globalObjectType || isEmptyObjectType(resolved)) ||
-                        resolved.stringIndexInfo || resolved.numberIndexInfo || getPropertyOfType(type, name)) {
+                    if ((relation === assignableRelation || relation === comparableRelation) && (type === globalObjectType || isEmptyObjectType(resolved)) ||
+                        resolved.stringIndexInfo ||
+                        getPropertyOfType(type, name)) {
                         return true;
                     }
+                    if (resolved.numberIndexInfo) {

```

Comment:
should not this be `(relation === assignableRelation || relation === comparableRelation) && resolved.numberIndexInfo`